docs/dev/cheatcode-mutability-review.md

```diff
@@ -0,0 +1,279 @@
+# Cheatcode Mutability Review Checklist
+
+This document tracks all cheatcodes currently marked `external` with no `view` or `pure` mutability modifier.
```

Comment:
I don't think merging this doc is useful so would remove. If you need to keep track, better to write a script or add a tasklist (though think 231 might be hard to maintain)